storage/pathmanager/pathManager_test.go

```diff
@@ -11,66 +12,82 @@ func TestNewPathManager_EmptyPruningPathTemplateShouldErr(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("", "shard_[S]/[I]", "db")
-	assert.Nil(t, pm)
+	assert.True(t, check.IfNil(pm))
 	assert.Equal(t, pathmanager.ErrEmptyPruningPathTemplate, err)
 }
 
 func TestNewPathManager_EmptyStaticPathTemplateShouldErr(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("epoch_[E]/shard_[S]/[I]", "", "db")
-	assert.Nil(t, pm)
+	assert.True(t, check.IfNil(pm))
 	assert.Equal(t, pathmanager.ErrEmptyStaticPathTemplate, err)
 }
 
+func TestNewPathManager_EmptyDBPathTemplateShouldErr(t *testing.T) {
+	t.Parallel()
+
+	pm, err := pathmanager.NewPathManager("epoch_[E]/shard_[S]/[I]", "shard_[S]/[I]", "")
+	assert.True(t, check.IfNil(pm))
+	assert.Equal(t, pathmanager.ErrInvalidDatabasePath, err)
+}
+
 func TestNewPathManager_InvalidPruningPathTemplate_NoShardPlaceholder_ShouldErr(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("epoch_[E]/shard/[I]", "shard_[S]/[I]", "db")
-	assert.Nil(t, pm)
+	assert.True(t, check.IfNil(pm))
 	assert.Equal(t, pathmanager.ErrInvalidPruningPathTemplate, err)
 }
 
 func TestNewPathManager_InvalidPruningPathTemplate_NoEpochPlaceholder_ShouldErr(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("epoch/shard_[S]/[I]", "shard_[S]/[I]", "db")
-	assert.Nil(t, pm)
+	assert.True(t, check.IfNil(pm))
 	assert.Equal(t, pathmanager.ErrInvalidPruningPathTemplate, err)
 }
 
 func TestNewPathManager_InvalidPathPruningTemplate_NoIdentifierPlaceholder_ShouldErr(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("epoch_[E]/shard_[S]", "shard_[S]/[I]", "db")
-	assert.Nil(t, pm)
+	assert.True(t, check.IfNil(pm))
 	assert.Equal(t, pathmanager.ErrInvalidPruningPathTemplate, err)
 }
 
 func TestNewPathManager_InvalidStaticPathTemplate_NoShardPlaceholder_ShouldErr(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("epoch_[E]/shard_[S]/[I]", "shard/[I]", "db")
-	assert.Nil(t, pm)
+	assert.True(t, check.IfNil(pm))
 	assert.Equal(t, pathmanager.ErrInvalidStaticPathTemplate, err)
 }
 
 func TestNewPathManager_InvalidStaticPathTemplate_NoIdentifierPlaceholder_ShouldErr(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("epoch_[E]/shard_[S]/[I]", "shard_[S]", "db")
-	assert.Nil(t, pm)
+	assert.True(t, check.IfNil(pm))
 	assert.Equal(t, pathmanager.ErrInvalidStaticPathTemplate, err)
 }
 
 func TestNewPathManager_OkValsShouldWork(t *testing.T) {
 	t.Parallel()
 
 	pm, err := pathmanager.NewPathManager("epoch_[E]/shard_[S]/[I]", "shard_[S]/[I]", "db")
-	assert.NotNil(t, pm)
+	assert.False(t, check.IfNil(pm))
```

Comment:
please revert to NotNil check

Reply:
reworked all tests from storage package. Left check.IfNil(pm) only when the tested function returned an interface.